relay-server/src/services/processor.rs

```diff
@@ -418,6 +418,9 @@ pub enum ProcessingError {
 
     #[error("invalid processing group type")]
     InvalidProcessingGroup(#[from] InvalidProcessingGroupType),
+
+    #[error("envelope invalid")]
+    Invalid(DiscardReason),
```

Comment:
Let's be a bit more explicit with the error as in the examples above, for example:
```suggestion
    #[error("invalid replay")]
    InvalidReplay(DiscardReason),
```
An ambiguous name could lead to making wrong assumptions when looking at errors.